test/state/host.cpp

```diff
@@ -228,9 +228,45 @@ std::optional<evmc_message> Host::prepare_message(evmc_message msg)
             else
             {
                 assert(msg.kind == EVMC_EOFCREATE);
-                const bytes_view initcontainer{msg.code, msg.code_size};
-                msg.recipient =
-                    compute_eofcreate_address(msg.sender, msg.create2_salt, initcontainer);
+                const bytes_view input = {msg.input_data, msg.input_size};
+
+                // Indicator of an EOF creation tx - EVMC_EOFCREATE at depth 0.
+                if (msg.depth == 0)
+                {
+                    // Assert this is a legacy EOF creation tx.
+                    assert(is_eof_container(input));
+
+                    const auto section_headers_or_error = validate_section_headers(input);
+                    if (holds_alternative<EOFValidationError>(section_headers_or_error))
+                        return {};  // Light early exception.
+
+                    const auto header = read_valid_eof1_header(input);
```

Comment:
it feels hacky to me, we don't read to re-parse the header at this point, we already have all section sizes returned from `validate_section_headers()` and can sum them up to find the split.

Reply:
I wanted to avoid the summing up, which we already handle in two places. BTW, did you see the comment https://github.com/ethereum/evmone/pull/878#issuecomment-2098375588, would this work in your opinion?

In general I would also prefer to take a safe and easy path now and possibly make a more thorough redesign of these validation routines, once we finalize the spec of the validations? There's still a lot of moving parts.